src/coreclr/jit/optimizemaskconversions.cpp

```diff
@@ -0,0 +1,532 @@
+    else if ((simdBaseJitType != newSimdBaseJitType) || (simdSize != newSimdSize))
+    {
+        // Type mismatch with existing cached type.
+        JITDUMP("Local V%02d has different types: (%d, %d) vs (%d, %d). ", lclnum, simdBaseJitType, simdSize,
+                newSimdBaseJitType, newSimdSize);
+        InvalidateWeight();
+    }
+}
+
+//-----------------------------------------------------------------------------
+// MaskConversionsCheckVisitor: Find all lcl var definitions and uses. For each one, update the weighting.
+//
+class MaskConversionsCheckVisitor final : public GenTreeVisitor<MaskConversionsCheckVisitor>
+{
+public:
+    enum
+    {
+        DoPostOrder       = true,
+        UseExecutionOrder = true
+    };
+
+    MaskConversionsCheckVisitor(Compiler* compiler, weight_t bbWeight, MaskConversionsWeightTable* weightsTable)
+        : GenTreeVisitor<MaskConversionsCheckVisitor>(compiler)
+        , bbWeight(bbWeight)
+        , weightsTable(weightsTable)
+    {
+    }
+
+    Compiler::fgWalkResult PostOrderVisit(GenTree** use, GenTree* user)
+    {
+        GenTreeHWIntrinsic* convertOp = nullptr;
+
+        bool isLocalStore  = false;
+        bool isLocalUse    = false;
+        bool isInvalid     = false;
+        bool hasConversion = false;
+
+        switch ((*use)->OperGet())
+        {
+            case GT_STORE_LCL_VAR:
+                isLocalStore = true;
+
+                // Look for:
+                //      use:STORE_LCL_VAR(ConvertMaskToVector(x))
+
+                if ((*use)->AsLclVar()->Data()->OperIsConvertMaskToVector())
+                {
+                    convertOp     = (*use)->AsLclVar()->Data()->AsHWIntrinsic();
+                    hasConversion = true;
+                }
+                break;
+
+            case GT_LCL_VAR:
+                isLocalUse = true;
+
+                // Look for:
+                //      user:ConvertVectorToMask(use:LCL_VAR(x)))
+
+                if (user->OperIsConvertVectorToMask())
+                {
+                    convertOp     = user->AsHWIntrinsic();
+                    hasConversion = true;
+                }
+                break;
+
+            default:
+                break;
+        }
+
+        if (isLocalStore || isLocalUse)
+        {
+            GenTreeLclVarCommon* lclOp  = (*use)->AsLclVarCommon();
+            LclVarDsc*           varDsc = m_compiler->lvaGetDesc(lclOp->GetLclNum());
+
+            // Get the existing weighting (if any).
+            MaskConversionsWeight  defaultWeight;
+            MaskConversionsWeight* weight = weightsTable->LookupPointerOrAdd(lclOp->GetLclNum(), defaultWeight);
+
+            JITDUMP("Local %s V%02d at [%06u] ", isLocalStore ? "store" : "var", lclOp->GetLclNum(),
+                    m_compiler->dspTreeID(lclOp));
+
+            // Cannot convert any locals with an exposed address.
+            if (varDsc->IsAddressExposed())
+            {
+                JITDUMP("is address exposed elsewhere. ");
```

Comment:
No, I'm intentionally not having the newlines, so that you end up with:

```
Local var V05 at [000041] is address exposed elsewhere. Invalidating weight. Weighting: Invalid{0.00c 0.00s}
Local store V00 at [000006] has no conversion. Incrementing switchCost by 20000.00. Weighting: {0.00c 20000.00s}
Local store V04 at [000007] has mask conversion. Incrementing currentCost by 10000.00. Weighting: {10000.00c 0.00s}
Local var V04 at [000009] has mask conversion. Incrementing currentCost by 20000.00. Weighting: {30000.00c 0.00s}
Local var V00 at [000010] is used as vector. Invalidating weight. Weighting: Invalid{0.00c 20000.00s}
```